src/matrix/room/RoomSummary.js

```diff
@@ -239,6 +239,15 @@ export class SummaryData {
         return !this.name && !this.canonicalAlias && this.heroes && this.heroes.length > 0;
     }
 
+    get avatarColorId() {
```

Comment:
Actually, I think your first idea to handle this in `Heroes` might be the right one. As you can see in `Invite.writeSync`, the heroes are derived from the invite state, as room invites never contain the `m.summary` section. So `this.heroes` will never be an array for an invite in the summary data. Given we already do the work of deriving the `Heroes`, better reuse that and put this property in there I think.